recipes/behaviortree.cpp/all/patches/4.0.2-0002-use-cci.patch

```diff
@@ -0,0 +1,56 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b713372..90fffec 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -59,9 +59,9 @@ endif()
+ 
+ #---- Find other packages ----
+ find_package(Threads REQUIRED)
+-find_package(ZMQ)
++find_package(ZeroMQ)
+ 
+-if( ZMQ_FOUND )
++if( ZeroMQ_FOUND )
+     message(STATUS "ZeroMQ: found.")
+     add_definitions( -DZMQ_FOUND )
+     list(APPEND BT_SOURCE src/loggers/bt_zmq_publisher.cpp)
+@@ -124,7 +124,10 @@ endif()
+ #############################################################
+ # LIBRARY
+ 
+-add_subdirectory(3rdparty/lexy)
++find_package(cppzmq REQUIRED CONFIG)
++find_package(lexy REQUIRED CONFIG)
++find_package(minitrace REQUIRED CONFIG)
++find_package(tinyxml2 REQUIRED CONFIG)
```

Comment:
These should be removed alson in 4.0.2 and 4.0.1:

- cppzmq
- lexy
- minitrace
- tinyxml2 

It was **never intended** to use conan for these specific dependencies, only sqlite and ZeroMQ

Reply:
The problem is that if the behaviortree library uses these dependencies internally, vendored, or something like that (it seems so https://github.com/BehaviorTree/BehaviorTree.CPP/tree/master/3rdparty), and the behaviortree library is used together with these vendored libraries, there could easily be linking errors, right? In principle, vendored libraries are not allowed in ConanCenter.

I'd like to understand a bit better about this.